src/main/java/dev/latvian/mods/kubejs/neoforge/NeoForgeEventsLookup.java

```diff
@@ -0,0 +1,206 @@
+					KubeJS.LOGGER.warn("[NativeEvents] Event class " + className + " for name '" + formattedName + "' is already registered as: " + eventMap.get(formattedName));
+					continue;
+				}
+
+				eventMap.put(formattedName, eventClass);
+			}
+		}
+
+		KubeJS.LOGGER.info("[NativeEvents] Loaded " + eventMap.size() + " events");
+	}
+
+	/**
+	 * Get all invalid classes which should not be loaded.
+	 * This includes all classes which are annotated with {@link Mixin} or with {@link OnlyIn}, when they current dist does not match.
+	 *
+	 * @param annotations all annotations
+	 * @return set of invalid classes
+	 */
+	private Set<String> getInvalidClasses(Set<ModFileScanData.AnnotationData> annotations) {
+		Type onlyIn = Type.getType(OnlyIn.class);
+		Type mixin = Type.getType(Mixin.class);
+
+		Set<String> invalid = new HashSet<>();
+		for (ModFileScanData.AnnotationData annotation : annotations) {
+			if (!annotation.targetType().equals(ElementType.TYPE)) {
+				continue;
+			}
+
+			if (isInvalidDist(annotation, onlyIn) || isMixin(annotation, mixin)) {
+				invalid.add(annotation.clazz().getClassName());
+			}
+		}
+
+		return invalid;
+	}
+
+	/**
+	 * Checks if given className is part of invalid classes. It also checks if the super class is part of invalid classes.
+	 *
+	 * @param invalidClasses set of invalid classes
+	 * @param className      class name
+	 * @return true if invalid
+	 */
+	private boolean isInvalidClass(Set<String> invalidClasses, String className) {
+		if (invalidClasses.contains(className)) {
+			return true;
+		}
+
+		String[] parts = className.split("\\$", 2);
+		return invalidClasses.contains(parts[0]);
+	}
+
+	private boolean isMixin(ModFileScanData.AnnotationData annotation, Type mixinType) {
+		return annotation.annotationType().equals(mixinType);
+	}
+
+	private boolean isInvalidDist(ModFileScanData.AnnotationData annotation, Type distType) {
+		Type type = annotation.annotationType();
+		if (!type.equals(distType)) {
+			return false;
+		}
+
+		Object unknownValue = annotation.annotationData().get("value");
+		if (unknownValue instanceof ModAnnotation.EnumHolder enumHolder) {
+			String invalidValue = FMLEnvironment.dist == Dist.CLIENT ? Dist.DEDICATED_SERVER.name() : Dist.CLIENT.name();
+			return enumHolder.getValue().equals(invalidValue);
+		}
+
+		return false;
+	}
+
+	/**
+	 * Checks if given class name is invalid. Only classes which end with "Event" are valid, but also subtypes of it.
+	 *
+	 * @param fullClassName class name
+	 * @return true if class name is invalid
+	 */
+	private boolean isInvalidClassName(String fullClassName) {
+		String[] parts = fullClassName.split("\\$");
+		for (String part : parts) {
+			if (part.endsWith("Event")) {
+				return false;
+			}
+		}
+
+		return true;
+	}
+
+	public String formatEventName(String className) {
+		var mainParts = className.split("\\$");
+		int index = 0;
+		for (int i = mainParts.length - 1; i >= 0; i--) {
+			if (mainParts[i].endsWith("Event")) {
```

Comment:
Again, relying on naming conventions, although at least here its only for displaying something to the user.

Reply:
Same as above. 

But this is just so that custom names just dont have `Event` in it. I preferred `NativeEvents.foo` instead of `NativeEvents.fooEvent`